datastore/src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -907,35 +853,57 @@ public Integer deleteEntityProperties(
     }
 
     @Timed
-    @RequestMapping(
-            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + NEIGHBORS },
-            method = RequestMethod.POST )
-    public Long deleteEntitiesAndNeighbors(
+    @Override
+    @DeleteMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + NEIGHBORS } )
+    public UUID deleteEntitiesAndNeighbors(
             @PathVariable( ENTITY_SET_ID ) UUID entitySetId,
             @RequestBody EntityNeighborsFilter filter,
             @RequestParam( value = TYPE ) DeleteType deleteType ) {
         // Note: this function is only useful for deleting src/dst entities and their neighboring entities
         // (along with associations connected to all of them), not associations.
         // If called with an association entity set, it will simplify down to a basic delete call.
 
-        ensureEntitySetCanBeWritten( entitySetId );
+        // filter.entityKeyIds should be non-empty
+        Preconditions.checkArgument( !filter.getEntityKeyIds().isEmpty(),
+                "EntityNeighborsFilter.entityKeyIds should be a non-empty set" );
+
+        Set<UUID> dstEntitySetIds = filter.getDstEntitySetIds().orElse( Set.of() );
+        Set<UUID> srcEntitySetIds = filter.getSrcEntitySetIds().orElse( Set.of() );
+        Set<UUID> allEntitySetIds = Sets.union( srcEntitySetIds, dstEntitySetIds );
+        allEntitySetIds = Sets.union( allEntitySetIds, Set.of( entitySetId ) );
 
-        WriteEvent writeEvent = new WriteEvent( 0, 0 );
+        ensureEntitySetsCanBeWritten( allEntitySetIds );
+
+        Map<UUID, Set<UUID>> entitySetIdEntityKeyIds = Maps.newHashMap();
+        entitySetIdEntityKeyIds.put( entitySetId, filter.getEntityKeyIds() );
+
+        // verify permission to delete
+        deletionManager.authCheckForEntitySetsAndNeighbors(
+                allEntitySetIds,
+                deleteType,
+                Principals.getCurrentPrincipals(), filter.getEntityKeyIds() );
+
+        UUID deletionJobId = deletionManager.clearOrDeleteEntitiesAndNeighbors(
```

Comment:
As far as I can tell, there has been no permissions check on the entity sets themselves up to this point. `ensureEntitySetsCanBeWritten` doesn't actually check permissions and `authCheckForEntitySetsAndNeighbors` only checks permissions on properties.

I think you  need to call `checkPermissionsOnEntitySetIds` instead of `ensureEntitySetsCanBeWritten`. However, this is not enough because it doesn't consider the edge entity sets.

Reply:
https://github.com/openlattice/openlattice/pull/254#discussion_r740719901